tests/functional/genesets/e2e_genesets.py

```diff
@@ -0,0 +1,38 @@
+import resdk
+
+from ..base import BaseResdkFunctionalTest
+
+
+class TestCreateGeneset(BaseResdkFunctionalTest):
+    def setUp(self):
+        super().setUp()
+        self.collection = self.res.collection.create(name="Test collection")
+        self.geneset = None
+
+    def tearDown(self):
+        # self.geneset is deleted along with collection
+        self.collection.delete(force=True)
+
+    def test_create_geneset(self):
+        self.geneset = self.res.geneset.create(
+            name="Test name",
+            genes=["FHIT", "MYC"],
+            source="UCSC",
+            species="Homo sapiens",
+        )
+
+        self.assertTrue(isinstance(self.geneset, resdk.resources.Geneset))
+        self.assertCountEqual(self.geneset.genes, ["FHIT", "MYC"])
+        self.assertEqual(self.geneset.source, "UCSC")
+        self.assertEqual(self.geneset.species, "Homo sapiens")
+        self.assertEqual(self.geneset.name, "Test name")
+
+        # Update geneset with a new collection and tag
+        self.geneset.collection = self.collection
+        self.geneset.collection.tags = ["community:expressions"]

```

Comment:
If you include collection argument in the `create` method, you can leave out collection here. In that case, assign tag to the geneset, not the collection.